design/milestone_1_2_design_doc.md

```diff
@@ -0,0 +1,430 @@
+
+Running as a **Job** allows separation from the applications' deployment and serve multiple applications at once. 
+
+Because for this Milestone we are concerned with updating K8s Secrets once at intial spin up, a **Job** is the most native solution. It will terminate upon task completion and not waste customer's resources.
+
+Kubernetes Helm is a tool that streamlines and organizes the management of Kubernetes installation, deployment, and upgrade processes. When we use Helm for deployment and lifecycle management, we can provide our customer's a one-click solution even though the way at which we deploy changes.
+
+#### What drawbacks does this solution have?
+
+**Job** is not an existing application identity granularity.
+
+ As a result, one can define the host representing the Secrets Provider in Conjur policy using only **namespace** and/or **service account** granularities.
+
+To handle the missing **Job** granularity there are 2 options:
+
+|      | Solution                                                     | Pros                                                         | Cons                                                         | Effort Estimation |
+| ---- | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------------------------------------------ | ----------------- |
+| 1    | Keep using only **namespace** and/or **service account** granularities. | \- Since Secrets Provider runs as a separate entity, we suggest creating a dedicated service account for Secrets Provider.<br />That way, the **service account** granularity serves as an exact match and **Job** granularity is redundant<br />- No code changes | \- Service account might be reused, which allows other apps to authenticate using Secrets Provider host<br />- Break application identity convention | Free              |
+| 2    | Add support for **Job** granularity in Conjur                | Allows specific identification of the Secrets Provider **Job**, which enhances security | \- Costly; requires code changes, verify compatibility, tests, docs<br />- Redundant if **service account** serves only the Secrets Provider | 10 days           |
+
+*Decision*: Solution #1, use **namespace** and **service account** to define the Secrets Provider host in Conjur policy. We would recommend to the customer to create a dedicated service account for Secrets Provider.
+
+#### Customer experience
+
+Deployment for the Secrets Provider will be done using Helm. That way, if the 
+
+1. Configure K8s Secrets with ***conjur-map\*** metadata defined
+
+2. Create `values.yaml` that will contain the following parameters and their defaults:
+
+   2.1. Service Account for Secrets Provider (default: `secrets-provider-account`)
+
+   2.2. Role (default: `secrets-provider-role`)
+
+   2.3. RoleBinding Name (default: `secrets-provider-role-binding`)
+
+   2.4. `K8S_SECRETS` (default: `[]`)
```

Comment:
please verify if other parameters are required - like conjur url and authenticator url

Reply:
Will K8S_SECRETS variable contain a full list of all secrets that will be synced ? If yes why do we need step 1 above ?

Reply:
they are, I just didn't detail each one of them... but you make a good point. I removed the list and added
> For a full list of required parameters, see `values.yaml` below

> Will K8S_SECRETS variable contain a full list of all secrets that will be synced ? If yes why do we need step 1 above 

What do you mean here? `K8S_SECRETS` are all the K8s secrets the user wants Conjur values for. Step 1, just says that inside the K8s secret, add conjur-map. Like so:
```
apiVersion: v1
kind: Secret
metadata:
  name: test-secret
type: Opaque
stringData:
  conjur-map: |-
    secret: secrets/test_secret
```
This is how we know the path to the Secret in Conjur